tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -0,0 +1,991 @@
+            )
+    # Cleanup no matter what
+    finally:
+        try:
+            team = zen_store.get_team(team_name_or_id=team_name)
+        except KeyError:
+            pass
+        else:
+            zen_store.delete_team(team.id)
+
+
+def test_removing_user_from_team_succeeds():
+    """Tests removing a user from a team."""
+
+    zen_store = Client().zen_store
+    team_name = sample_name("arias_team")
+
+    with UserContext() as created_user:
+        try:
+            new_team = TeamRequestModel(
+                name=team_name, users=[created_user.id]
+            )
+            new_team = zen_store.create_team(new_team)
+            assert created_user.id in new_team.user_ids
+
+            team_update = TeamUpdateModel(users=[])
+            team_update = zen_store.update_team(
+                team_id=new_team.id, team_update=team_update
+            )
+
+            assert created_user.id not in team_update.user_ids
+        # Cleanup no matter what
+        finally:
+            try:
+                team = zen_store.get_team(team_name_or_id=team_name)
+            except KeyError:
+                pass
+            else:
+                zen_store.delete_team(team.id)
+
+
+#  .------.
+# | USERS |
+# '-------'
+
+
+def test_active_user():
+    """Tests the active user can be queried with .get_user()."""
+    zen_store = Client().zen_store
+    active_user = zen_store.get_user()
+    assert active_user is not None
+    # The SQL zen_store only supports the default user as active user
+    if zen_store.type == StoreType.SQL:
+        assert active_user.name == DEFAULT_USERNAME
+    else:
+        # TODO: Implement this
+        assert True
+
+
+def test_updating_default_user_fails():
+    """Tests that updating the default user is prohibited."""
+    client = Client()
+    default_user = client.zen_store.get_user(DEFAULT_USERNAME)
+    assert default_user
+    user_update = UserUpdateModel(name="axl")
+    with pytest.raises(IllegalOperationError):
+        client.zen_store.update_user(
+            user_id=default_user.id, user_update=user_update
+        )
+
+
+def test_deleting_default_user_fails():
+    """Tests that deleting the default user is prohibited."""
+    zen_store = Client().zen_store
+    with pytest.raises(IllegalOperationError):
+        zen_store.delete_user("default")
+
+
+# .-------.
+# | ROLES |
+# '-------'
+
+
+def test_creating_role_with_empty_permissions_succeeds():
+    """Tests creating a role."""
+    zen_store = Client().zen_store
+    new_role = RoleRequestModel(name=sample_name("cat"), permissions=set())
+    created_role = zen_store.create_role(new_role)
+    with does_not_raise():
+        zen_store.get_role(role_name_or_id=new_role.name)
+    list_of_roles = zen_store.list_roles(RoleFilterModel(name=new_role.name))
+    assert list_of_roles.total > 0
+    # Cleanup
+    with does_not_raise():
+        zen_store.delete_role(created_role.id)
```

Comment:
RoleContext should help with this, but maybe you're testing a corner case where this actually doesn't make sense. In that case, maybe a try/catch ?